lib/resources/user-collection.js

```diff
@@ -98,14 +97,25 @@ UserCollection.prototype.handle = function (ctx) {
           return noSuchUser();
         }
 
-        ctx.query = {id: ctx.session.data.uid};
+        ctx.query = ctx.query || {};
+        ctx.query.id = ctx.session.data.uid;
+
+        // raw store query, need unmodified username and password
+        return uc.store.find({id: ctx.session.data.uid, $fields: {username: 1, password: 1}}, function(err, user) {
+          if (err) {
+            console.error("Internal Error: " + JSON.stringify(err));
+            return ctx.done({statusCode: 500, message: "Error retrieving user for verification"});
```

Comment:
I'm hiding the actual error here from the response for security purposes as it may potentially contain sensitive data.